File: 021/021.py
def sum_divisors(n):
    dsum = 0
    i = 2
    while i*i < n:
        if n % i == 0:
            dsum += i + (n // i)
        i += 1
    if i*i == n:
        dsum += i
    return dsum + 1


amicable_sum = 0
for a in range(2, 1000):
    b = sum_divisors(a)
    if b > a:
        if sum_divisors(b) == a:
            amicable_sum += a + b

print(amicable_sum)

File: solutions/021/021.go
package main

import "fmt"

func sumDivisors(n int) int {
	sum := 0
	i := 2
	for ; i*i < n; i++ {
		if n%i == 0 {
			sum += i + (n / i)
		}
	}
	if i*i == n {
		sum += i
	}
	return sum + 1
}

func main() {
	amicableSum := 0
	for a := 2; a < 10000; a++ {
		b := sumDivisors(a)
		if (b > a) && (sumDivisors(b) == a) {
			amicableSum += a + b
		}
	}
	fmt.Println(amicableSum)
}
